java/src/main/native/src/row_conversion.cu

```diff
@@ -1251,10 +1331,12 @@ static inline int32_t compute_fixed_width_layout(std::vector<data_type> const &s
  * @return size of the fixed_width data portion of a row.
  */
 template <typename iterator>
-static size_type compute_column_information(iterator begin, iterator end,
-                                            std::vector<size_type> &column_starts,
-                                            std::vector<size_type> &column_sizes) {
-  size_type fixed_width_size_per_row = 0;
+static column_info_s compute_column_information(iterator begin, iterator end) {
```

Comment:
Nitpick: The `static` might be superfluous here, since this is a function template.